argus/action_manager/windows.py

```diff
@@ -169,19 +170,43 @@ def sysprep(self):
         self.wait_boot_completion()
 
     def git_clone(self, repo_url, location):
-        """Clone from an remote repo to a specific location on the instance.
+        """Clone a remote repository to a specific location on the instance.
 
         :param repo_url:
-            The remote repo url.
+            The remote repository url.
         :param location:
             Specific location on the instance.
         """
-        LOG.info("Cloning from %s to %s", repo_url, location)
-        cmd = "git clone {} {}".format(repo_url, location)
-        self._client.run_command_with_retry(cmd,
-                                            count=util.RETRY_COUNT,
-                                            delay=util.RETRY_DELAY,
-                                            command_type=util.CMD)
+        LOG.info('Trying to clone from %r to "%s".', repo_url, location)
+        cmd = 'git clone {repo_url} "{location}"'.format(
+            repo_url=repo_url, location=location)
+
+        count = 0
+
+        while True:
+            try:
+                if self.exists(location):
+                    if self.is_dir(location):
+                        self.rmdir(location)
+                    else:
+                        self.remove(location)
+
+                LOG.info("Cloning the repository.")
```

Comment:
Don't think we need to log each clone try since you already log the cloning on line 180.